### ISSUE TITLE
BUG - A schema rule could not be found after sufficient iterations

### ISSUE THREAD

**Issue by Pennywise83** [Foundry376/Mailspring#805]
Hi,
when I read a very long thread with some attachments, mailspring crashes and show me the following error:

Error: A schema rule could not be validated after sufficient iterations. This is usually due to a `rule.validate` or `rule.normalize` function of a schema being incorrectly written, causing an infinite loop.
    at iterate (/usr/share/mailspring/resources/app.asar/node_modules/slate/lib/changes/with-schema.js:178:13)
    at iterate (/usr/share/mailspring/resources/app.asar/node_modules/slate/lib/changes/with-schema.js:182:5)
    at iterate (/usr/share/mailspring/resources/app.asar/node_modules/slate/lib/changes/with-schema.js:182:5)
    at iterate (/usr/share/mailspring/resources/app.asar/node_modules/slate/lib/changes/with-schema.js:182:5)
    at iterate (/usr/share/mailspring/resources/app.asar/node_modules/slate/lib/changes/with-schema.js:182:5)

I'm unable to open that thread after that, and I have to restart mailspring (avoiding to click on the "affected" thread in order to use it).

**Comment by lampo808** [comment]
Happening very frequently also to me.

**Comment by tgalopin** [comment]
Happening to me all the time, this makes Mailspring unusable for me :/ .

**Comment by lestevam** [comment]
It happened to me too in an email with many posts.

**Comment by pyeguy** [comment]
getting this randomly, (on pressing reply in this case)
Email : G-Suite
Mailspring: 1.5 (pro) 
OS:  Ubuntu 18.04

```
Error: A schema rule could not be validated after sufficient iterations. This is usually due to a `rule.validate` or `rule.normalize` function of a schema being incorrectly written, causing an infinite loop.
    at iterate (/snap/mailspring/259/usr/share/mailspring/resources/app.asar/node_modules/slate/lib/changes/with-schema.js:178:13)
    at iterate (/snap/mailspring/259/usr/share/mailspring/resources/app.asar/node_modules/slate/lib/changes/with-schema.js:182:5)
    at iterate (/snap/mailspring/259/usr/share/mailspring/resources/app.asar/node_modules/slate/lib/changes/with-schema.js:182:5)
    at iterate (/snap/mailspring/259/usr/share/mailspring/resources/app.asar/node_modules/slate/lib/changes/with-schema.js:182:5)
    at iterate (/snap/mailspring/259/usr/share/mailspring/resources/app.asar/node_modules/slate/lib/changes/with-schema.js:182:5)
```

**Comment by bergmul** [comment]
Happened to me also twice already. Could only solve it be reinstalling the app. I had it both on Ubuntu 18.10 with the snap version and on Elementary Juno beta 2 with the non-snap version.

Email: IMAP
Mailspring: 1.5.2 (snap)
OS: Ubuntu 18.10

**Comment by bengotow** [comment]
Related to #921. This is being prioritized for the next release, sorry for the delay folks. Unfortunately this SlateJS text editor is causing us a lot of grief. Will see what we can do.

**Comment by CodeMouse92** [comment]
It appears that this issue has been resolved, so we are closing it. **If the problem occurs again, please create a new thread in the appropriate category on our [Discourse](https://community.getmailspring.com).**

Resolved here: 21bc2ef398e403ab064f7046240b2fcff455a5e2

We are in the process of migrating issues to Discourse, which can better facilitate discussion and discovery, and so GitHub Issues can focus on issues that are confirmed and slated for resolution in the near term. [Learn more about the changes here.](https://community.getmailspring.com/t/mailspring-community-transition/38)

We hope to see you there soon!

-The Mailspring Team
